atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -168,6 +169,20 @@ boolean allowHiddenTableAccess() {
 
     abstract TaggedMetricRegistry globalTaggedMetricRegistry();
 
+    /**
+     * If config().initializeAsync() is set to true, the callback Runnable will be run when the TransactionManager is
+     * successfully initialized. The TransactionManager stay uninitialized and continue to throw for all other purposes
+     * until the callback returns at which point it will become initialized.
+     *
+     * Note that if the callback blocks forever, the TransactionManager will never become initialized. If the callback
```

Comment:
If the callback blocks forever on a single iteration it will also be impossible to close the TM as the close method will block indefinitely.